Src/BootCamp.Chapter/Command/CityCommand.cs

```diff
@@ -0,0 +1,111 @@
+﻿using BootCamp.Chapter.ReportsManagers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    internal class CityCommand : ICommand
+    {
+        private string _Path;
+        private List<string> _Command;
+        private List<Transaction> _Transactions;
+        private ReportsManager _ReportsManager;
+
+        public CityCommand(string path, List<string> command, List<Transaction> transactions, ReportsManager reportsManager)
+        {
+            _Path = path;
+            _Command = command;
+            _Transactions = transactions;
+            _ReportsManager = reportsManager;
+        }
+
+        public void Execute()
+        {
+            var toBeWritten = CreateReport();
+            WriteToFile(toBeWritten);
+        }
+        private IEnumerable<string> CreateReport()
+        {
+            const string money = "-money";
+            const string items = "-items";
+            const string min = "-min";
+            const string max = "-max";
+            string command = _Command[0];
+            string commandParameter1 = _Command[1];
+            string commandParameter2 = _Command[2];
+
+            if (_Command.Count != 3)
+            {
+                throw new InvalidCommandException($"{_Command.ToString()} has the wrong amount of parameters.");
+            }
+
+            if (commandParameter1 == items)
+            {
+                if (commandParameter2 == min || commandParameter2 == max)
+                {
+                    return FindItemsMaxOrMin(_Transactions, commandParameter2);
+                }
+                else
+                {
+                    throw new InvalidCommandException($"{commandParameter2} is not a valid parameter.");
+                }
+            }
+            else if (commandParameter1 == money)
+            {
+                if (commandParameter2 == min || commandParameter2 == max)
```

Comment:
if min find mind, if max find max. I think this is overengineering trying to do both things at once.